src/frontend/src/pages/ProjectPage/pages/ContractPage/components/DelegateAccessSideSheet/useNewDelegation.ts

```diff
@@ -0,0 +1,79 @@
+import { PersonDetails, useCurrentContext, useNotificationCenter } from '@equinor/fusion';
+import * as React from 'react';
+import {
+    PersonDelegationClassification,
+    PersonDelegationRequest,
+} from '../../../../../../models/PersonDelegation';
+import { useAppContext } from '../../../../../../appContext';
+import { useContractContext } from '../../../../../../contractContex';
+
+export default (
+    toDate: Date | null,
+    persons: PersonDetails[],
+    accountType: PersonDelegationClassification,
+    onSuccessfullyDelegated?: () => void,
+) => {
+    const { apiClient } = useAppContext();
+    const { contract, dispatchContractAction } = useContractContext();
+    const currentContext = useCurrentContext();
+
+    const [isDelegatingAccess, setIsDelegatingAccess] = React.useState<boolean>(false);
+    const [delegateError, setDelegateError] = React.useState<Error | null>(null);
+    const sendNotification = useNotificationCenter();
+
+    const delegateAccessAsync = React.useCallback(
+        async (projectId: string, contractId: string, validTo: Date) => {
+            setIsDelegatingAccess(true);
+            setDelegateError(null);
+            try {
+                const response = persons.map(async (person) => {
```

Comment:
dont think async is required here, since api client returns a promise

Reply:
True